language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -60,12 +64,21 @@ tests damlc daml2ts davl = testGroup "daml2Ts"
       buildProject grover []
       assertBool "grover-1.0.dar was not created." =<< doesFileExist groverDar
       step "Generating TypeScript of 'grover'..."
-      daml2tsProject [groverDar] groverTs
-      assertBool "'Grover.ts' was not created." =<< doesFileExist (groverTs </> "grover-1.0" </> "Grover.ts")
-      assertBool "'packageId.ts' was not created." =<< doesFileExist (groverTs </> "grover-1.0" </> "packageId.ts")
+      daml2tsProject [groverDar] (groverTs </> "src")
+      assertBool "'Grover.ts' was not created." =<< doesFileExist (groverTs </> "src" </> "grover-1.0" </> "Grover.ts")
+      assertBool "'packageId.ts' was not created." =<< doesFileExist (groverTs </> "src" </> "grover-1.0" </> "packageId.ts")
+      step "Compiling 'grover-ts' to JavaScript... "
+      writeTsConfig groverTs
+      writePackageJson groverTs "@breathing-test/grover"
+      -- Invoke 'yarn install'. We only need to do this once.
```

Comment:
once per what? once per test? once for all tests? If it’s once for all tests I would move it out of this test otherwise this is going to break if you filter out tests with `-p`.

Reply:
The idea is to `yarn install` just once for all tests. I think, will have to play, this requires of course a config file exists for it do read dependencies from. I'll see what I can do about moving it out.

Reply:
Makes sense, I would factor this out into a tasty resource then. Then it will only be run once but it will be guaranteed to run even if you filter out tests.

Reply:
We did some work on this. It's better now.